code/controllers/subsystem/fire_burning.dm

```diff
@@ -27,10 +27,12 @@ SUBSYSTEM_DEF(fire_burning)
 				return
 			continue
 
-		if(O.resistance_flags & ON_FIRE)
-			O.take_damage(20, BURN, "fire", 0)
-		else
-			processing -= O
```

Comment:
I think this particular line wasn't doing anything, though, as extinguish() already removes O from processing.

Reply:
correct